### PR TITLE
[dashboard] Add throttling to IDB persist in PersistQueryClientProvider – WEB-560

### PR DESCRIPTION
**Pull request by AlexTugarev**
As [described in the initial discussion](https://github.com/TanStack/query/discussions/1638) on IndexedDB backed cache for react-query, writes to IDB should be throttled to "avoid indexedDB spamming". 

With this PR writes to IDB are throttles, i.e. there might be writes missed out between the leading and the trailing edge, but the most recent update is expected to happen. 

Why is this significant? react-query is aggressively calling `persistClient` function on each change of each query. We've seen in debugger, that on dehydration (on load) it will re-add each query's and persist in-between. Having multiple tabs of Gitpod open at the same time, this might case a blocking state with IDB. Most of those persist calls are not required, only the last update is complete and awaited on, so that the query can continue.

## Related Issue(s)

Fixes WEB-560

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-first-api-call</li>
	<li><b>🔗 URL</b> - <a href="https://at-first-api-call.preview.gitpod-dev.com/workspaces" target="_blank">at-first-api-call.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/hooks/use-user-loader.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export const useUserLoader = () => {
     const { isLoading } = useQuery({
         queryKey: noPersistence(["current-user"]),
         queryFn: async () => {
+            console.timeLog("first-api-call");
```
Comment (easyCZ) [inline components/dashboard/src/hooks/use-user-loader.ts:26]:
```suggestion
```

File: components/dashboard/src/index.tsx [hunk 2]
```diff
@@ -82,4 +82,5 @@ const bootApp = () => {
     );
 };
 
+console.time("first-api-call");
```
Comment (easyCZ) [inline components/dashboard/src/index.tsx:85]:
```suggestion
```

File: components/dashboard/src/user-settings/Integrations.test.tsx_
Comment (easyCZ) [inline components/dashboard/src/user-settings/Integrations.test.tsx_:1]: revert change here

Comment (AlexTugarev) [reply components/dashboard/src/user-settings/Integrations.test.tsx_:1]: There is a problem with module loading of p-throttle, so it breaks jest.

**Review by easyCZ — APPROVED** [review]

File: components/dashboard/src/user-settings/Integrations.test.tsx_
Comment (AlexTugarev) [reply components/dashboard/src/user-settings/Integrations.test.tsx_:1]: I wanted to rescue the test file, to have a blueprint for how to write test for react components. But I failed to fix module loading issues. The test case itself is no longer needed, as the value it's checking for is no longer derived from the entered host, but it's a fixed value now. I've commented on the [commit](https://github.com/gitpod-io/gitpod/pull/18082/commits/b80e26c79dc2f880463a7fc950431a6bd3a8e899) accordingly.

Happy to try to revive that blue print in a follow up.

File: components/dashboard/src/index.tsx (hunk 2 above)
Comment (AlexTugarev) [reply components/dashboard/src/index.tsx]: cleaned

File: components/dashboard/src/hooks/use-user-loader.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/dashboard/src/hooks/use-user-loader.ts]: cleaned

File: components/dashboard/src/user-settings/Integrations.test.tsx_
Comment (easyCZ) [reply components/dashboard/src/user-settings/Integrations.test.tsx_:1]: it's in git history, if needed

**Pull request merged**